src/molecules/top-navigation-bar.tsx

```diff
@@ -21,30 +27,58 @@ export type TopNavBarProps = {
 
 const DEFAULT_LABEL = "ホーム";
 
-const TopNavBar = ({
-  subjects,
-  userInitial,
-  label = DEFAULT_LABEL,
-}: TopNavBarProps) => {
+const TopNavBar = ({ userInitial, label = DEFAULT_LABEL, subjects }: TopNavBarProps) => {
   const [isSideMenuOpen, setIsSideMenuOpen] = useState(false);
-
   const openSideMenu = () => setIsSideMenuOpen(true);
   const closeSideMenu = () => setIsSideMenuOpen(false);
 
+  const [isAccountMenuOpen, setIsAccountMenuOpen] = useState(false);
+  const toggleAccountMenu = () => setIsAccountMenuOpen((flag) => !flag);
+  
+  const menuRef = useRef<HTMLDivElement>(null);
+  const user: { name: string; initials: string; } | null = null;
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (menuRef.current && !menuRef.current.contains(event.target as HTMLElement)) {
+        setIsAccountMenuOpen(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+  
   return (
     <>
       <div className={`${styles.topNavBar} surface on-surface-text`}>
         <MdMenu className={styles.menuButton} onClick={openSideMenu} />
         <CurrentScreenLabel label={label} />
-        <UserAvatar userInitial={userInitial} />
+        <UserAvatar userInitial={userInitial} onClick={toggleAccountMenu} />
+      </div>
+      {isSideMenuOpen && (
+        <SideMenu
+          isOpen={isSideMenuOpen}
+          closeMenu={closeSideMenu}
+          subjects={subjects}
+        />
+      )}
+      {isAccountMenuOpen && (
+        <div ref={menuRef}>
+        <AccountMenu
+          isLoggedIn={user !== null}
+          userName={user?.name}
+          userInitial={user?.initials}
+          onLogout={() => console.log('ログアウト処理')}
+          onLogin={() => console.log('ログイン処理')}
+        />
```

Comment:
```suggestion
        <AccountMenu
          ref={menuRef}
          user={user}
          onLogout={() => console.log('ログアウト処理')}
          onLogin={() => console.log('ログイン処理')}
        />
```
これだけでいいはずです (指定できなかったのですがこの後ろの行の `</div>` も消しましょう).